module/VuFind/src/VuFind/Db/Table/UserResource.php

```diff
@@ -164,8 +164,12 @@ public function destroyLinks($resource_id, $user_id, $list_id = null)
         // Remove any tags associated with the links we are removing; we don't
         // want to leave orphaned tags in the resource_tags table after we have
         // cleared out favorites in user_resource!
-        $this->getDbService(ResourceTagsServiceInterface::class)
-            ->destroyResourceTagsLinksForUser($resource_id, $user_id, $list_id);
+        $resourceTagsService = $this->getDbService(ResourceTagsServiceInterface::class);
+        if ($list_id === true) {
```

Comment:
I'll replace the destroyLinks method here with better named and better clarified methods in the next stage of refactoring, but I want to get this piece settled first.